### PR TITLE
diffuse 0.7.3 (new formula)

### PR DESCRIPTION
**Pull request by chenrui333**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

I have test these functionalities in my local, but I dont know how to test out that in the GUI setup, so just checking the version and help instead.
```
Compare a Set of Files
$ diffuse file1 file2 file3

Inspect a Revision
$ diffuse -c rev
```

continuing #96392
relates to https://github.com/MightyCreak/diffuse/pull/145

### CODE REVIEW

**Comment by chenrui333** [comment]
I am going to close this PR since upstream has already moved into [making diffuse an native app](https://github.com/MightyCreak/diffuse/pull/165)

**Comment by chenrui333** [comment]
It should be better put into homebrew-cask, asking [in the upstream](https://github.com/MightyCreak/diffuse/issues/187) now

File: Formula/diffuse.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  homepage "https://mightycreak.github.io/diffuse/"
+  url "https://github.com/MightyCreak/diffuse/archive/refs/tags/v0.7.3.tar.gz"
+  sha256 "cceff03beabc83fc76c49ecdad6743b7bd709063167af851fe2ebae7a10b1394"
+  license "GPL-2.0-or-later"
+
+  depends_on "gettext" => :build
+  depends_on "meson" => :build
+  depends_on "ninja" => :build
+
+  depends_on "gtk+3"
```
Comment (hugoholgersson) [inline Formula/diffuse.rb:12]: When I studied Diffuse on macOS, I found that we actually need a few other deps, see:
[brew install meson python3 py3cairo pygobject3 gtk+3 librsvg](https://github.com/MightyCreak/diffuse/blame/master/docs/developers/developers-setup.md#L45)

**Comment by hugoholgersson** [comment]
> I am going to close this PR since upstream has already moved into [making diffuse an native app](https://github.com/MightyCreak/diffuse/pull/165)

@chenrui333, even though Diffuse now feels more like a native app, we think it would still make sense to have Diffuse in `brew` for easy installation:

> [Regular users should have simpler ways to install Diffuse. ... On macOS, ... Homebrew.](https://github.com/MightyCreak/diffuse/pull/179#issuecomment-1322541506)

 Brew already contains apps like Gimp, which is another example of an app that behaves as a native app.

File: Formula/diffuse.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    ENV["DESTDIR"] = "/"
+
+    system "meson", *std_meson_args, "build"
+    system "meson", "compile", "-C", "build", "-v"
+    system "meson", "install", "-C", "build"
+  end
+
+  test do
+    assert "Diffuse is a graphical tool for merging and comparing text files",
+      shell_output("#{bin}/diffuse --help")
```
Comment (hugoholgersson) [inline Formula/diffuse.rb:26]: As of today, this needs to be `diffuse_impl`, not `diffuse`.

See discussion in https://github.com/MightyCreak/diffuse/pull/179#discussion_r1011474671.

File: Formula/diffuse.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
+class Diffuse < Formula
+  desc "Graphical tool for merging and comparing text files"
+  homepage "https://mightycreak.github.io/diffuse/"
+  url "https://github.com/MightyCreak/diffuse/archive/refs/tags/v0.7.3.tar.gz"
```
Comment (hugoholgersson) [inline Formula/diffuse.rb:4]: Let's use the lastest version?

**Comment by MightyCreak** [comment]
@chenrui333: I agree with @hugoholgersson that Diffuse could still be added to Homebrew. Thanks to @hugoholgersson's work Diffuse is now better integrated in macOS, but it's not part of any Mac app store (this would be a whole other task! :sweat_smile: ).

**Pull request closed without merging**